relayer/chains/evm/client.go

```diff
@@ -19,34 +19,33 @@ import (
 const (
 	DefaultPollingInterval = time.Second * 30
 	MaximumPollTry         = 15
+	DefaultCreateTimeout   = time.Second * 10
 )
 
 func newClient(ctx context.Context, connectionContract, XcallContract common.Address, rpcUrl, websocketUrl string, l *zap.Logger) (IClient, error) {
-	rpc, err := ethclient.DialContext(ctx, rpcUrl)
+	createCtx, cancel := context.WithTimeout(ctx, DefaultCreateTimeout)
```

Comment:
Can be renamed to ctx?

Reply:
no, it cannot since client creation required longer context, this context is only for creating clients are call for getting chainId.